src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs

```diff
@@ -48,16 +48,36 @@ internal class ConditionalTestDiscoverer
                     continue;
                 }
 
-                string[] symbols = conditionMemberName.Split('.');
                 Type declaringType = testMethodDeclaringType;
 
-                if (symbols.Length == 2)
+                // We have qualified type name with the assembly name, something like
+                // [ConditionalFact("System.PlatformDetection, CoreFx.Private.TestUtilities!" + nameof(PlatformDetection.IsNonZeroLowerBoundArraySupported))]
+                // We don't use '.' as separator in such case and use '!' because the qualified type name can have '.' to include the type namespace.
+                if (conditionMemberName.IndexOf('!') > 0)
                 {
-                    conditionMemberName = symbols[1];
-                    ITypeInfo type = testMethod.TestClass.Class.Assembly.GetTypes(false).Where(t => t.Name.Contains(symbols[0])).FirstOrDefault();
-                    if (type != null)
+                    // get the method name
+                    string[] symbols = conditionMemberName.Split('!');
+                    if (symbols.Length == 2)
```

Comment:
Right now if  it couldn't find the method it will return an ExecutionErrorTestCase which means that the test will not be discovered and will print an error saying that the property didn't exist. 

https://github.com/dotnet/buildtools/blob/master/src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs#L65-L75

So we don't really throw exceptions we print an error message for that specific test through xunit (which I think is better) 

This is the error message that would be displayed: https://github.com/dotnet/buildtools/blob/master/src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs#L105-L108